src/Raven.Server/Documents/Handlers/Admin/AdminLogsHandler.cs

```diff
@@ -77,5 +81,47 @@ public async Task RegisterForLogs()
                 await LoggingSource.Instance.Register(socket, context, ServerStore.ServerShutdown);
             }
         }
+
+        [RavenAction("/admin/logs/download", "GET", AuthorizationStatus.Operator)]
+        public async Task Download()
+        {
+            var contentDisposition = $"attachment; filename={DateTime.UtcNow:yyyy-MM-dd H:mm:ss} - Node {ServerStore.NodeTag} - Logs.zip";
+            HttpContext.Response.Headers["Content-Disposition"] = contentDisposition;
+
+            using (ServerStore.ContextPool.AllocateOperationContext(out JsonOperationContext context))
+            {
+                await using (var ms = new MemoryStream())
+                {
+                    using (var archive = new ZipArchive(ms, ZipArchiveMode.Create, true))
+                    {
+                        foreach (var filePath in Directory.GetFiles(ServerStore.Configuration.Logs.Path.FullPath))
+                        {
+                            try
+                            {
+                                using (var fs = File.Open(filePath, FileMode.Open, FileAccess.Read, FileShare.ReadWrite))
+                                {
+                                    var fileName = Path.GetFileName(filePath);
+                                    var entry = archive.CreateEntry(fileName);
+
+                                    entry.ExternalAttributes = ((int)(FilePermissions.S_IRUSR | FilePermissions.S_IWUSR)) << 16;
+
+                                    await using (var entryStream = entry.Open())
+                                    {
+                                        await fs.CopyToAsync(entryStream);
+                                    }
+                                }
+                            }
+                            catch
+                            {
+                                // something went wrong with copying the file, will continue to the next one
```

Comment:
should we create fake zip entry with errors in such case?

Reply:
yes

Reply:
done